src/turndown_test.js

```diff
@@ -180,11 +180,29 @@ _italics wrapped in a div_
     })
   })
 
+  it("should properly convert 1 liner code snippet into its correct markdown", async () => {
+    const inputHTML = `<span style="font-family: Courier New,Courier;">sudo apt-get install -y python2.7 python-profiler</span>`
+    const markdown = await html2markdown(inputHTML)
+    assert.equal(
+      markdown,
+      "`sudo apt-get install -y python2.7 python-profiler`"
+    )
+  })
+
+  it("should properly convert code block into its correct markdown", async () => {
+    const inputHTML = `<pre>import time<br>print('I am going to sleep :)')<br>time.sleep(1000)</pre>`
+    const markdown = await html2markdown(inputHTML)
+    assert.equal(
+      markdown,
+      "```import time  \nprint('I am going to sleep :)')  \ntime.sleep(1000)```"
+    )
+  })
+
   it("should not get tripped up on problematic code blocks", async () => {
     const problematicHTML =
       "<pre><span><code>stuff\nin\nthe\nblock</span></pre>"
     const markdown = await html2markdown(problematicHTML)
-    assert.equal(markdown, "```\nstuff\nin\nthe\nblock\n```")
+    assert.equal(markdown, "````stuff\nin\nthe\nblock````")
```

Comment:
This test was failing and I also didn't want to change the function so I had to update it's comparison value here, not sure whether its correct.
@alicewriteswrongs Did you write this? If yes, then is it correct or I did something wrong?

Reply:
@MAbdurrehman12 This was introduced in [this](https://github.com/mitodl/ocw-to-hugo/pull/58) PR.  I think that if you're seeing this test fail, the original problem this was meant to solve is not being solved anymore.  I think the reason for this is that your turndown rule simply filters on all `pre` elements whereas the other rule applies to `pre` elements that also have a `span` as their first child.  It would seem that your rule that simply filters on `pre` is being selected instead of the other rule.  What I would do here is combine both sets of logic into one turndown rule.  The filter for the turndown rule itself should only be `node.nodeName === "PRE"`, and then in the replacement rule apply the logic from the replacement rule on `codeblockfix` if `node.firstChild && node.firstChild.nodeName === "SPAN"` is true.  This would ensure that only one rule is followed in both scenarios.

Reply:
Thanks for pointing this out @gumaerc. I'll update the PR